controllers/replication.storage/volumereplication_controller.go

```diff
@@ -421,7 +421,10 @@ func getScheduleTime(parameters map[string]string, logger logr.Logger) time.Dura
 		logger.Error(err, "failed to parse time: %v", rawScheduleTime)
 		return defaultScheduleTime
 	}
-	return scheduleTime
+	// Reduce the schedule time by half to get the latest update and also to
+	// avoid the inconsistency between the last sync time in the VR and the
+	// Storage system.
+	return scheduleTime / 2
```

Comment:
Shouldn't this be done by the entity creating VRC in the first place?

I don't think it is logical to to add this logic at this level.

Reply:
Please explain why you dont think this logic should not be added here. Please check the commit message for the proper explanation of why this is done.

Reply:
>Reduce the schedule time by half to get the latest update and also to avoid the inconsistency between the last sync time in
the VR and the Storage system.

SchedulingTime from the VRC implies the time interval between which we will schedule a re-queue to check for updates right ? 
Can the reducing the time by half done at a above level ? 

>The user can see updates for RPO that are not stuck in a bad schedule race, i.e, VolumeReplication checks and finds sync time as t-5m, and just after that storage system updates it to t+x. If we checked every 1/2 of the schedule we will update it to t+x in t+s/2

By checking 1/2 of the schedule, we are not obeying the provided scheduling time at all? 
Is there a reason that the entity handling this cannot create VRC with half of the required shedule?

Reply:
>SchedulingTime from the VRC implies the time interval between which we will schedule a re-queue to check for updates right ?
Can the reducing the time by half done at a above level ?

This is not true, This is the time specific to the storage, what is the interval to take the snapshot of the rbd image(for example ceph). we are here using the same time to Requeue to check the LastSyncTime/Update time of the replication.

>By checking 1/2 of the schedule, we are not obeying the provided scheduling time at all?
Is there a reason that the entity handling this cannot create VRC with half of the required schedule?

Please Check the above explanation, the schedule interval is not meant for GetReplicationInfo Requeue time it's for the Storage Replication interval time, we are using the same time to get the updates on the replication time. if we keep both the same, there is a kind of diff between the VR LastSyncTime and the storage LastSyncTime. to keep both the same, we are decreasing the time by half so that we can keep the LastSyncTime and the storageLastSyncTime to almost the same.

Reply:
okay, 

https://github.com/csi-addons/kubernetes-csi-addons/blob/main/docs/volumereplicationclass.md
Can we add description here and how it used in parameters section passed to csi-driver
And for requeue time of reconcile
and mention it in function description too .

Reply:
done, ptal